src/operator/timeout.ts

```diff
@@ -45,6 +46,8 @@ class TimeoutOperator<T> implements Operator<T, T> {
 class TimeoutSubscriber<T> extends Subscriber<T> {
   private index: number = 0;
   private _previousIndex: number = 0;
+  private action: Subscription = null;
+
   get previousIndex(): number {
```

Comment:
If we're tracking the scheduled action subscription, do we still need `previousIndex` etc.? Same question for `hasCompleted`.